packages/react-reconciler/src/ReactFiberCommitWork.new.js

```diff
@@ -2639,6 +2664,7 @@ function commitMutationEffectsOnFiber(
       }
 
       commitReconciliationEffects(finishedWork);
+      finishedWork.stateNode._current = finishedWork;
```

Comment:
Add a todo to make sure there's an explicit effect flag scheduled to set this. Currently we're assuming that if the fiber is cloned, we'll visit it in the mutation phase, but that's not necessarily true in all cases even if it happens to mostly be true today due to overtraversing.

Reply:
Let's address that in a follow up PR.